chapters/intro/intro.md

```diff
@@ -0,0 +1,108 @@
+
+Inspired by Brown and Wilsons' [Architecture of Open Source Applications][aosa], we decided to organize each description as a chapter, resulting in the present on line book.
+
+## Recurring Themes
+
+The chapters share several common themes, which are based on smaller assignments the students conducted as part of the course. These themes cover different architectural 'theories' as available on the web or in textbooks. The course used  Rozanski and Woods' [Software Systems Architecture][rw], and therefore several of their architectural [viewpoints] and [perspectives] recur.
+
+[viewpoints]: http://www.viewpoints-and-perspectives.info/home/viewpoints/
+[perspectives]: http://www.viewpoints-and-perspectives.info/home/perspectives/
+
+The first theme is outward looking, focusing on the use of the system. Thus, many of the chapters contain an explicit [stakeholder analysis], as well as a description of the [context] in which the systems operate. These were based on available on line documentation, as well as on an analysis of open and recently closed issues for these systems.
+
+[context]: http://www.viewpoints-and-perspectives.info/home/viewpoints/context/
+[stakeholder analysis]: http://www.mindtools.com/pages/article/newPPM_07.htm
+
+A second theme involves the [development viewpoint][development], covering modules, layers, components, and their inter-dependencies. Furthermore, it addresses integration and testing processes used for the system under analysis.
+
+[development]: http://www.viewpoints-and-perspectives.info/home/viewpoints/
+
+A third recurring theme is _variability management_. Many of today's software systems are highly configurable. In such systems, different features can be enabled or disabled, at compile time or at run time. Using techniques from the field of [product line engineering][fospl], several of the chapters provide feature-based variability models of the systems under study.
+
+A fourth theme is [metrics-based evaluation of software architectures][bouwers]. Using such metrics architects can discuss  (desired) quality attributes (performance, scaleability, maintainability, …) of a system quantitatively. Therefore various chapters discuss metrics and in some cases actual measurements tailored towards the systems under analysis.
+
+
+## First-Hand Experience
+
+Last but not least, the chapters are also based on the student's experience in actually contributing to the systems described. As part of the course over 75 pull requests to the projects under study were made, including refactorings
+([Jekyll 3545](https://github.com/jekyll/jekyll/pull/3545),
+  [Docker 11350](https://github.com/docker/docker/pull/11350),
+  [Syncany 391](https://github.com/syncany/syncany/pull/391)), bug fixes
+([Docker 11018](https://github.com/docker/docker/pull/11018),
```

Comment:
Yea, that's definately not a bugfix. This one [Docker 11085](https://github.com/docker/docker/pull/11085) is a bugfix, but it wasn't merged as is, because the problem turned out much more difficult. However, the commits were used to fix it eventually. 
[Docker 11323](https://github.com/docker/docker/pull/11323) is indeed a nice example of a refactoring that was done to make Docker compileable for Windows.